org/mozilla/jss/ssl/javax/JSSEngine.java

```diff
@@ -0,0 +1,781 @@
+     * It is suggested that at least one key manager be a JSSKeyManager
+     * instance if a key and certificate must be provided for this end of
+     * the connection.
+     */
+    public void setKeyManagers(X509KeyManager[] xkms) {
+        if (xkms == null) {
+            logger.debug("JSSEngine: setKeyManagers([null])");
+            return;
+        }
+
+        logger.debug("JSSEngine: setKeyManagers(");
+        for (X509KeyManager km : xkms) {
+           logger.debug(" - " + km.getClass().getName());
+        }
+        logger.debug(")");
+
+        key_managers = xkms;
+    }
+
+    /**
+     * Set the internal TrustManager, when present, replacing all previous
+     * TrustManagers.
+     */
+    public void setTrustManager(JSSTrustManager tm) {
+        if (tm == null) {
+            logger.debug("JSSEngine: setTrustManager(null)");
+            return;
+        }
+
+        logger.debug("JSSEngine: setTrustManager(" + tm.getClass().getName() + ")");
+        trust_managers = new X509TrustManager[] { tm };
+    }
+
+    /**
+     * Set the internal list of TrustManagers.
+     */
+    public void setTrustManagers(X509TrustManager[] xtms) {
+        if (xtms == null) {
+            logger.debug("JSSEngine: setKeyManagers([null])");
+            return;
+        }
+
+        logger.debug("JSSEngine: setTrustManagers(");
+        for (X509TrustManager tm : xtms) {
+           logger.debug(" - " + tm.getClass().getName());
+        }
+        logger.debug(")");
+
+        trust_managers = xtms;
+    }
+
+    /**
+     * Gets the JSSSession object which reflects the status of this
+     * JSS Engine's session.
+     */
+    public JSSSession getSession() {
+        logger.debug("JSSEngine: getSession()");
+        return session;
+    }
+
+    /**
+     * Whether or not to enable this SSLEngine instance to create new
+     * sessions.
+     *
+     * The default value is true. When passed the value false, this will
+     * throw a RuntimeException, stating that all JSS Engines do not support
+     * restricting to only resuming existing sessions.
+     */
+    public void setEnableSessionCreation(boolean flag) {
+        logger.debug("JSSEngine: setEnableSessionCreation(" + flag + ") - not implemented");
+        if (!flag) {
+            String msg = "JSSEngine does not support restricting to only resuming existing sessions.";
+            throw new RuntimeException(msg);
+        }
+    }
+
+    /**
+     * Whether or not new sessions can be created by this SSLEngine instance.
+     *
+     * This always returns true.
+     */
+    public boolean getEnableSessionCreation() {
+        logger.debug("JSSEngine: getEnableSessionCreation() - not implemented");
+        return true;
+    }
+
+    /**
+     * Set whether or not to handshake as a client.
+     */
+    public void setUseClientMode(boolean mode) throws IllegalArgumentException {
+        logger.debug("JSSEngine.setUseClientMode(" + mode + ")");
+        is_client = mode;
```

Comment:
Usually the field name should match the setter/getter methods (i.e. `useClientMode`). Something for other fields below.

Reply:
I think I want to rename these to match the NSS versions, not the SSLEngine versions. i.e., the internals are all NSS names for things (`asServer`) and then the NSS portion makes sense and only the translation layer at the SSLEngine interface has to care about the naming of things.